executor/load_data.go

```diff
@@ -98,9 +95,84 @@ func (e *LoadDataExec) Open(ctx context.Context) error {
 	if e.loadDataInfo.insertColumns != nil {
 		e.loadDataInfo.initEvalBuffer()
 	}
+	dataStream := e.ctx.Value(LoadDataInput)
+	if dataStream == nil {
+		dataStream = ctx.Value(LoadDataInput)
+		if dataStream == nil {
+			return errors.New("ctx has no LoadDataInput data stream")
+		}
+		e.ctx.SetValue(LoadDataInput, dataStream)
+	}
 	return nil
 }
 
+// DoCmd do the load data work, input DataStream could
+// be clientConn or read file on server(non local mode)
+func (e *LoadDataExec) DoCmd(ctx context.Context) error {
+	var err error
+	loadDataInfo := e.loadDataInfo
+	// LoadDataInput default is server.clientConn,
+	// if non local mode is supported in the future,
+	// LoadDataInput value in Ctx should be reset in buildLoadData
+	input, ok := e.ctx.Value(LoadDataInput).(DataStream)
+	if !ok {
+		return errors.New("load data input is nil")
+	}
+	err = input.LoadPreCheck()
+	if err != nil {
+		return err
+	}
+	err = input.ReqData(loadDataInfo.Path)
+	if err != nil {
+		return err
+	}
+
+	var shouldBreak bool
+	var prevData, curData []byte
+	err = loadDataInfo.Ctx.NewTxn(ctx)
+	if err != nil {
+		return err
+	}
+	for {
+		curData, err = input.ReadFromStream()
+		if err != nil {
+			if terror.ErrorNotEqual(err, io.EOF) {
+				logutil.Logger(ctx).Error("input failed", zap.Error(err))
+				break
+			}
+		}
+		if len(curData) == 0 {
+			shouldBreak = true
```

Comment:
The `shouldBreak` looks complex...

```
for {
    curData, err = input.ReadFromStream()
    if err != nil {
       ... 
    }

    if len(data) == 0 {
       continue
    }
    ...
}
```

Reply:
This is because later left `PrevData` need to be processed. so mark here after processing, then break loop.
Actually this pr does not change load data logic related code (they are refactored in #11533 using two execution modes, but #11533 is not merged and seems a littble big), so split the "traceable logic into " this pr to speed up review.

What bothering me is that, in old logic clientConn.handleQuery will do work `DoCmd` function here do, so we need to pass clientConn related things here. I tried to pass this as `DataStream`
interface during the BuildExecutor phase(or other execution method phase), but there seems to be recursive import problem , is there some better way to solve this ? @jackysp @tiancaiamao